public/javascripts/SVLabel/src/SVLabel/keyboard/Keyboard.js

```diff
@@ -226,73 +181,17 @@ function Keyboard (svl, canvas, contextMenu, googleMap, ribbon, zoomControl) {
              */
             status.shiftDown = e.shiftKey;
             if (!status.focusOnTextField) {
-                switch (e.keyCode) {
-                    // Label selection hotkeys
-                    case util.misc.getLabelDescriptions('Occlusion')['shortcut']['keyNumber']:
-                        // "b" for a blocked view.
-                        // Context menu may be open for a different label.
-                        _closeContextMenu(e.keyCode);
-                        ribbon.modeSwitch("Occlusion");
-                        svl.tracker.push("KeyboardShortcut_ModeSwitch_Occlusion", {
+                // e: Walk, c: CurbRamp, m: NoCurbRamp, o: Obstacle, s: SurfaceProblem: n: NoSidewalk, o: Occlusion
+                for (const mode of ['Walk', 'CurbRamp', 'NoCurbRamp', 'Obstacle', 'SurfaceProblem', 'NoSidewalk', 'Occlusion']) {
+                    if (e.keyCode === util.misc.getLabelDescriptions(mode)['shortcut']['keyNumber']) {
+                        if (mode !== 'Walk') _closeContextMenu(e.keyCode);
```

Comment:
I also added in this check. We were not previously closing the context menu when you pressed 'e' for explore. Not sure what the original rationale was (possibly that you are already put back in explore mode after you place a label and the context menu opens up). But the reason it's important now is that we need the letter 'e' for our tag keyboard shortcuts. Instead of adding the 'tree' tag it was just closing the context menu.